pycbc/events/events.py

```diff
@@ -546,8 +547,288 @@ def add_template_events_to_ifo(self, ifo, columns, vectors):
+        assert new_events is not None
+        new_events['template_id'] = self.template_index
+        for c, v in zip(columns, vectors):
+            if v is not None:
+                if isinstance(v, Array):
+                    new_events[c] = v.numpy()
+                else:
+                    new_events[c] = v
+        self.template_events = numpy.append(self.template_events, new_events)
+
+    def add_template_events_to_network(self, columns, vectors):
+        """ Add a vector indexed """
+        # Just call through to the standard function
+        self.template_events = self.template_event_dict['network']
+        self.add_template_network_events(columns, vectors)
+        self.template_event_dict['network'] = self.template_events
+        self.template_events = None
+
+    def write_to_hdf(self, outname):
+        class fw(object):
+            def __init__(self, name):
+                import h5py
+                self.f = h5py.File(name, 'w')
+
+            def __setitem__(self, name, data):
+                col = self.prefix + '/' + name
+                self.f.create_dataset(col, data=data,
+                                      compression='gzip',
+                                      compression_opts=9,
+                                      shuffle=True)
+
+        self.events.sort(order='template_id')
+        th = numpy.array([p['tmplt'].template_hash for p in \
+                          self.template_params])
+        tid = self.events['template_id']
+        f = fw(outname)
+        #Output network stuff
+        f.prefix = 'network'
+        network_events = numpy.array([e for e in self.network_events],
+                                     dtype=self.network_event_dtype)
+        f['event_id'] = network_events['event_id']
+        f['network_snr'] = network_events['network_snr']
+        f['null_snr'] = network_events['null_snr']
+        f['end_time_gc'] = network_events['time_index'] / \
+                float(self.opt.sample_rate[self.ifos[0].lower()]) + \
+                        self.opt.gps_start_time[self.ifos[0].lower()]
+        f['nifo'] = network_events['nifo']
+        f['latitude'] = network_events['latitude']
+        f['longitude'] = network_events['longitude']
+        f['template_id'] = network_events['template_id']
+        for ifo in self.ifos:
+            #First add the ifo event ids to the network branch
+            f[ifo + '_event_id'] = network_events[ifo + '_event_id']
+        #Individual ifo stuff
+        for i, ifo in enumerate(self.ifos):
+            f.prefix = ifo
+            ifo_events = numpy.array([e for e in self.events if \
+                    e['ifo'] == self.ifo_dict[ifo]], dtype=self.event_dtype)
+            if len(ifo_events):
+                ifo_str = ifo.lower()[0] if ifo != 'H1' else ifo.lower()
+                f['snr_%s' % ifo_str] = abs(ifo_events['snr'])
+                f['event_id'] = ifo_events['event_id']
+                try:
+                    # Precessing
```

Comment:
Precessing? This approach wouldn't apply directly to a coherent search, unfortunately.

Reply:
Should I just delete this then? I just left it in because it was there for the multi det event manager.